### PR TITLE
Task/ghgc 140 viewer changes for new masked dataset

### PR DESCRIPTION
**Pull request by sanzog03**

### CODE REVIEW

**Review by paridhi-parajuli — APPROVED** [review]

**Review by ram-sharan25 — APPROVED** [review]

File: goes-plume-viewer/src/assets/dataset/metadata.ts [hunk 1]
```diff
@@ -1,168 +1,216 @@
-export const PLUMES_META = {
-    "BV1-1": {
+import { PlumeMeta } from "../../dataModel";
+
+
+
+export const PlumeMetas: PlumeMeta[] = [
```
Comment (slesaad) [inline goes-plume-viewer/src/assets/dataset/metadata.ts:5]: is this not available in the web somewhere?

File: goes-plume-viewer/src/components/colorBar/helper/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+function generateScale(min, max, step) {
+    const numbers = [];
+    for (let i = min; i <= max; i += step) {
+        numbers.push(i);
+    }
+    numbers[numbers.length - 1] += "+";
+    return numbers;
+}
+
```
Comment (slesaad) [inline goes-plume-viewer/src/components/colorBar/helper/index.js:45]: 👀

File: goes-plume-viewer/src/components/drawer/index.jsx [hunk 3]
```diff
@@ -122,17 +137,25 @@ export function PersistentDrawerRight({open, setOpen, selectedPlume, collectionI
...
-              endDatetime={selectedPlumeMeta.endDatetime}
-            />
+          { !!selectedPlumeMetas.length &&
+            selectedPlumeMetas.map(selectedPlumeMeta => (
+              <PlumeCard
+                key={selectedPlumeMeta.id}
+                plumeSourceId={selectedPlumeMeta.id}
+                plumeSourceName={selectedPlumeMeta.id.replace(/_/g, " ")}
+                imageUrl={`${process.env.REACT_APP_RASTER_API_URL}/collections/${collectionId}/items/${plumesMap[selectedPlumeMeta.id].representationalPlume.id}/preview.png?assets=rad&rescale=${VMIN}%2C${VMAX}&colormap_name=${colorMap}`}
+                tiffUrl={`${process.env.REACT_APP_RASTER_API_URL}/collections/${collectionId}/items/${plumesMap[selectedPlumeMeta.id].representationalPlume.id}/preview.png?assets=rad&rescale=${VMIN}%2C${VMAX}&colormap_name=${colorMap}`}
```
Comment (slesaad) [inline goes-plume-viewer/src/components/drawer/index.jsx:147]: why two of the same thing? 🤔

Comment (sanzog03) [reply goes-plume-viewer/src/components/drawer/index.jsx:154]: will change that once i get the tillUrl. It will be something different.

Comment (sanzog03) [reply goes-plume-viewer/src/components/drawer/index.jsx:154]: will be working on it in a different branch.

File: goes-plume-viewer/src/utils/trieTree.js [hunk 4]
```diff
@@ -21,7 +21,11 @@ export class TrieTree{
             node = node.children[letter]
         }
         node.isCompleteWord = true;
-        node.word = word;
+        if (alternateWord) {
+            node.word = alternateWord;
+        } else {
+            node.word = word;
+        }
```
Comment (slesaad) [inline goes-plume-viewer/src/utils/trieTree.js:28]: 👀

Comment (sanzog03) [reply goes-plume-viewer/src/utils/trieTree.js:28]: a little enhancement on typical trie tree. i.e. for each word in the plume Id, we should show the result for the plumeId and not that word completion. for instance, Typing in `Duran` (for Durango) in search should recommend Mexico_Durango_BV1_BV1-1 and not just Durango.

**Review by slesaad — APPROVED** [review]

**Pull request merged**